File: 0345-reverse-vowels-of-a-string/0345-reverse-vowels-of-a-string.py
class Solution:
    def reverseVowels(self, s: str) -> str:
        if len(s) == 1:
            return s
        s = list(s) 
        left = 0
        right = len(s) - 1
        while left < right:
            if s[left] not in "aeiouAEIOU":
                left += 1
            if s[right] not in "aeiouAEIOU":
                right -= 1
            if s[left] in "aeiouAEIOU" and s[right] in "aeiouAEIOU":
                s[left], s[right] = s[right], s[left]
                left += 1
                right -= 1
        return "".join(s)
        
            

File: 0345-reverse-vowels-of-a-string/0345-reverse-vowels-of-a-string.cpp
class Solution {
    bool isVowel(char c){
            if(c == 'A' || c == 'a' || c == 'e' || c == 'E' || c == 'i' || c == 'I' || c == 'o' || c == 'O' || c == 'u' || c == 'U'){
                return true;
            }
            return false;
        }
public:
    string reverseVowels(string s) {
        int l = 0;
        int r = s.size() - 1;
        while(l < r){
            if(isVowel(s[l]) && isVowel(s[r])){
                char temp = s[l];
                s[l] = s[r];
                s[r] = temp;
                l++;
                r--;
            }else if(!isVowel(s[l])){
                l++;
            }else if(!isVowel(s[r])){
                r--;
            }
        }
        return s;
    }
};